### DISCUSSION TITLE
Astro: only generate route in dev mode

### DISCUSSION THREAD

**Discussion by florian-lefebvre** [Thinkmill/keystatic#486] [Q&A] [answered]
Hi there! I'm using keystatic to manage my content, but only locally. But when building the app, the /keystatic route is still accessible even if useless. So I'm wondering if there's a way to keep it in dev mode only.
I doubt it's doable right now but in the future, an idea could be to just ship an astro integration and manage it with vite inject route utilities

**Comment by simonswiss** [comment, marked as answer]
I haven't tried this, but wondering if you could use [Astro.redirect](https://docs.astro.build/en/core-concepts/routing/#dynamic-redirects) to redirect to the homepage when `import.meta.env.MODE` is set to `production`?

```diff
// src/pages/keystatic/[...params].astro
---
import { Keystatic } from '../../../keystatic.page'

export const prerender = false

+ if (import.meta.env.MODE === 'production') return Astro.redirect('/')
---

<Keystatic client:only />
```

Disabling access to Keystatic in prod is definitely a valid idea — I have heard a few other request it (or simply worry about the concept of folks accessing admin routes in production).

**Reply by florian-lefebvre** [reply]
Will try this thanks! I'll keep you updated of course 👍

**Reply by simonswiss** [reply]
How did you go with this? Is it a viable workaround?

**Reply by florian-lefebvre** [reply]
Does the trick thank you! I don't know about seo (using `@astrojs/sitemap`), it probably needs [filtering](https://docs.astro.build/en/guides/integrations-guide/sitemap/#filter).

Haven't tried it yet but simething like this sounds reasonable:
```ts
// ...
sitemap({
  filter: (page) => !page.includes("keystatic"),
});
```

**Reply by florian-lefebvre** [reply]
I think it's worth documenting, may I submit a PR?

**Reply by simonswiss** [reply]
To the Astro integration guide? Yes please!

**Reply by needim** [reply]
What about disabling it completely on production env?

<img width="771" alt="image" src="https://github.com/Thinkmill/keystatic/assets/271912/f3338563-edbe-4623-8876-c042cbe92955">

**Reply by florian-lefebvre** [reply]
That was before the integration existed but I guess that should work now!

**Reply by needim** [reply]
Cool, it looks like the docs need an update.
